File: Second Year/2nd Semester/Medii de Proiectare si Programare/C#/Laborator_2/repository/RefereeDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Runtime.InteropServices;
using Laborator_1.domain;
using log4net;

namespace Laborator_1.repository
{
    public class RefereeDBRepository:RefereeRepository
    {
        private static readonly ILog log = LogManager.GetLogger("RefereeDBRepository");
        private IDictionary<string, string> props;
        
        public RefereeDBRepository(IDictionary<string, string> props)
        {
            log.Info("Creating RefereeDBRepository"); 
            this.props = props;
        }
        
        public Referee FindOne(int id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection conn = DBUtils.getConnection(props);
            using (var command = conn.CreateCommand())
            {
                command.CommandText = "select id,username,password,first_name,last_name,activity  from Referees where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);

                using (var dataR = command.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idV = dataR.GetInt32(0);
                        String username = dataR.GetString(1);
                        String password = dataR.GetString(2);
                        String first_name = dataR.GetString(3);
                        String last_name = dataR.GetString(4);
                        String activity = dataR.GetString(5);
                        Referee referee = new Referee(id, username, password, first_name, last_name, activity);
                        log.InfoFormat("Existing findOne with value {0}", referee);
                        return referee;
                    }
                }
            }
            log.InfoFormat("Existing findOne with value {0}", null);
            return null;
        }

        public IEnumerable<Referee> FindAll()
        {
            log.Info("Find ALL Referees");
            IDbConnection conn = DBUtils.getConnection(props);
            IList<Referee> referees = new List<Referee>();
            using (var command = conn.CreateCommand())
            {
                command.CommandText = "select id,username,password,first_name,last_name,activity  from Referees";
                using (var dataR = command.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        String username = dataR.GetString(1);
                        String password = dataR.GetString(2);
                        String first_name = dataR.GetString(3);
                        String last_name = dataR.GetString(4);
                        String activity = dataR.GetString(5);
                        Referee referee = new Referee(id, username, password, first_name, last_name, activity);
                        log.InfoFormat("Existing find Referee with value {0}", id);
                        referees.Add(referee);
                    }
                }
            }

            return referees;
        }

        public void Save(Referee entity)
        {
            var conn = DBUtils.getConnection(props);
            using (var command = conn.CreateCommand())
            {
                command.CommandText =
                    "insert into Referees values(@id, @username, @password, @first_name, @last_name, @activity)";
                var paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = entity.Id;
                command.Parameters.Add(paramId);
                
                var paramUsername = command.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = entity.Username;
                command.Parameters.Add(paramUsername);
                
                var paramPassword = command.CreateParameter();
                paramPassword.ParameterName = "@password";
                paramPassword.Value = entity.Password;
                command.Parameters.Add(paramPassword);
                
                var paramFirstName = command.CreateParameter();
                paramFirstName.ParameterName = "@first_name";
                paramFirstName.Value = entity.First_Name;
                command.Parameters.Add(paramFirstName);
                
                var paramLastName = command.CreateParameter();
                paramLastName.ParameterName = "@last_name";
                paramLastName.Value = entity.Last_Name;
                command.Parameters.Add(paramLastName);
                
                var paramActivity = command.CreateParameter();
                paramActivity.ParameterName = "@activity";
                paramActivity.Value = entity.Activity;
                command.Parameters.Add(paramActivity);

                var result = command.ExecuteNonQuery();
                if (result == 0)
                    throw new RepositoryException("No referee added!");
            }
        }

        public void Delete(int id)
        {
            IDbConnection conn = DBUtils.getConnection(props);
            using (var command = conn.CreateCommand())
            {
                command.CommandText = "delete from Referees where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);
                var dataR = command.ExecuteNonQuery();
                if (dataR == 0)
                    throw new RepositoryException("No referee deleted!");
            }
        }

        public void Update(Referee entity)
        {
            throw new System.NotImplementedException();
        }

        public List<Referee> findByActivity(string Activity)
        {
            log.InfoFormat("Entering findByActivity with value {0}", Activity);
            IDbConnection conn = DBUtils.getConnection(props);
            List<Referee> referees = new List<Referee>();
            using (var command = conn.CreateCommand())
            {
                command.CommandText = "select * from Referees where activity=@activity";
                IDbDataParameter paramActivity = command.CreateParameter();
                paramActivity.ParameterName = "@activity";
                paramActivity.Value = Activity;
                command.Parameters.Add(paramActivity);

                using (var dataR = command.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        String username = dataR.GetString(1);
                        String password = dataR.GetString(2);
                        String first_name = dataR.GetString(3);
                        String last_name = dataR.GetString(4);
                        String activity = dataR.GetString(5);
                        Referee referee = new Referee(id, username, password, first_name, last_name, activity);
                        log.InfoFormat("Existing findByActivity with balue {0}",Activity);
                        referees.Add(referee);
                    }
                }
            }
            log.InfoFormat("Existing findbyActivity with value {0}",Activity);
            return referees;
        }
    }
}

File: Second Year/2nd Semester/Medii de Proiectare si Programare/Java/Laborator_1/src/main/java/repository/RefereeDBRepository.java
package repository;

import java.sql.*;
import java.util.ArrayList;

import domain.Referee;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.List;
import java.util.Properties;

public class RefereeDBRepository implements RefereeRepository{
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();

    public RefereeDBRepository(Properties props){
        logger.info("Initializing RefereeDBRepository with propertiesL {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public Referee findOne(Integer i) throws IllegalArgumentException {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();
        Referee referee = null;
        try(PreparedStatement preSmt= conn.prepareStatement("select * from Referees where id=?")){
            preSmt.setInt(1,i);
            try(ResultSet resultSet = preSmt.executeQuery()){
                if(!resultSet.next())
                    return null;

                int id = resultSet.getInt("id");
                String username = resultSet.getString("username");
                String password = resultSet.getString("password");
                String first_name = resultSet.getString("first_name");
                String last_name = resultSet.getString("last_name");
                String activity = resultSet.getString("activity");

                referee = new Referee(id,username,password,first_name,last_name,activity);
            }
        }catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit(referee);
        return referee;
    }

    @Override
    public Iterable<Referee> getAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Referee> refereeList = new ArrayList<>();
        try(PreparedStatement preSmt=con.prepareStatement("select * from Referees")){
            try(ResultSet resultSet = preSmt.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");
                    String username = resultSet.getString("username");
                    String password = resultSet.getString("password");
                    String first_name = resultSet.getString("first_name");
                    String last_name = resultSet.getString("last_name");
                    String activity = resultSet.getString("activity");

                    Referee referee = new Referee(id,username,password,first_name,last_name,activity);
                    refereeList.add(referee);
                }
            }
        } catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit(refereeList);
        return refereeList;
    }

    @Override
    public void clear() {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();
        try(PreparedStatement preSmt= conn.prepareStatement("delete from Referees")){
            preSmt.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit();
    }

    @Override
    public void save(Referee entity) {
        logger.traceEntry("saving task{} ", entity);
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Referees (username,password,first_name,last_name,activity) values (?,?,?,?,?)")){
            preStmt.setString(1,entity.getUsername());
            preStmt.setString(2,entity.getPassword());
            preStmt.setString(3,entity.getFirst_name());
            preStmt.setString(4,entity.getLast_name());
            preStmt.setString(5,entity.getActivity());
            int result=preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Referee entity) {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();
        try(PreparedStatement preSmt= conn.prepareStatement("update Referees set username = ?, password = ?, first_name = ?, last_name = ?, activity = ? where id=?")){
            preSmt.setString(1,entity.getUsername());
            preSmt.setString(2,entity.getPassword());
            preSmt.setString(3,entity.getFirst_name());
            preSmt.setString(4,entity.getLast_name());
            preSmt.setString(5,entity.getActivity());
            preSmt.setInt(6,entity.getId());
            preSmt.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit(entity);
    }

    @Override
    public void delete(Integer id) {
        logger.traceEntry();
        Connection conn = dbUtils.getConnection();
        try(PreparedStatement preSmt= conn.prepareStatement("delete from Referees where id=?")){
            preSmt.setInt(1,id);
            preSmt.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit(id);
    }



    @Override
    public List<Referee> findByActivity(String activity) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Referee> refereeList = new ArrayList<>();
        try(PreparedStatement preSmt=con.prepareStatement("select * from Referees where activity=?")){
            preSmt.setString(1,activity);
            try(ResultSet resultSet = preSmt.executeQuery()){
                while(resultSet.next()){
                    int id = resultSet.getInt("id");
                    String username = resultSet.getString("username");
                    String password = resultSet.getString("password");
                    String first_name = resultSet.getString("first_name");
                    String last_name = resultSet.getString("last_name");
                    String activityR = resultSet.getString("activity");

                    Referee referee = new Referee(id,username,password,first_name,last_name,activityR);
                    refereeList.add(referee);
                }
            }
        } catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB "+e);
        }
        logger.traceExit(refereeList);
        return refereeList;
    }
}
